src/drive.c

```diff
@@ -923,6 +929,7 @@ int ilm_scsi_list_rescan(void)
 		ret = ilm_read_device_wwn(dev_node, &wwn);
 		if (ret < 0) {
 			ilm_log_err("fail to read parttable id");
+			ret = 0;
```

Comment:
How about we add the sentence "ret = 0" after the looping and before calling ilm_scsi_dump_nodes().

```
        for (i = 0; i < num; ++i) {
        ...
        }

        /*
         * FIXUP: When scan the drive list, though it might have some failures for specific drives,
         * but we still want the system to continue to run.  So always return 0.
         */  
        ret = 0;

        ilm_scsi_dump_nodes();
```